cowprotocol/services
# Adjust order limit prices for volume based fee

# Description
Fixes https://github.com/cowprotocol/services/issues/2503

# Changes
Limit prices are adjusted after being scaled (for partial fillable orders) so that the order limit prices sent to solvers are WORSE than real ones, so that the driver has more room to withhold the fee on solution without violating limit prices.

## How to test
Existing tests.
e2e tests `local_node_volume_fee_sell_order` and `local_node_volume_fee_buy_order`. No change in the expectation which is good. Observed locally that we actually send adjusted sell/buy limit amounts to solvers but the solution is the same, since it does not affect the baseline solver.

For driver tests, expectation on what limit amounts are sent from driver to solver are changed, which is expected.
